src/main/java/thePackmaster/actions/doppelpack/EmergencyContactAction.java

```diff
@@ -0,0 +1,50 @@
+package thePackmaster.actions.doppelpack;
+
+import basemod.helpers.CardModifierManager;
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import thePackmaster.cardmodifiers.doppelpack.ShowDoppel;
+
+import java.util.ArrayList;
+
+public class EmergencyContactAction extends AbstractGameAction {
+    private final String text;
+    private ArrayList<AbstractCard> affectedCards;
+
+    public EmergencyContactAction(int amount, String description) {
+        this.amount = amount;
+        this.text = description;
+    }
+
+    @Override
+    public void update() {
+        isDone = true;
+        affectedCards = new ArrayList<>(AbstractDungeon.player.drawPile.group);
+        for (AbstractCard card : affectedCards) {
+            CardModifierManager.addModifier(card, new ShowDoppel());
+        }
+
+        addToBot(new SelectFromGridAction(
+                (s, c) -> s == SelectFromGridAction.Source.DRAW_PILE,
+                this::afterSelected,
+                text,
+                ActionType.CARD_MANIPULATION,
+                amount,
+                false
+        ));
+    }
+
+    private void afterSelected(SelectFromGridAction.Source[] sources, AbstractCard[] abstractCards) {
+        if (affectedCards != null) {
```

Comment:
Under what circumstances can affected cards be null? Should be an empty list at worst whcih works fine in a for-each loop

Reply:
This is updated now to make sure it's always non-null.